examples/CampaignManagement/AddCompleteCampaignsUsingMutateJob.php

```diff
@@ -0,0 +1,486 @@
+            ['pageSize' => self::PAGE_SIZE]
+        );
+        foreach ($mutateJobResults->iterateAllElements() as $mutateJobResult) {
+            /** @var MutateJobResult $mutateJobResult */
+            printf(
+                "Mutate job #%d has a status '%s' and response '%s'.%s",
+                $mutateJobResult->getOperationIndex(),
+                $mutateJobResult->getStatus()
+                    ? $mutateJobResult->getStatus()->getMessage() : 'N/A',
+                $mutateJobResult->getMutateOperationResponse()
+                    ? $mutateJobResult->getMutateOperationResponse()->getResponse()
+                    : 'N/A',
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Builds all operations for creating a complete campaign and return an array of their
+     * corresponding mutate operations.
+     *
+     * @param int $customerId the client customer ID
+     * @return MutateOperation[] the mutate operations to be added to a mutate job
+     */
+    private static function buildAllOperations(int $customerId)
+    {
+        $mutateOperations = [];
+
+        // Creates a new campaign budget operation and add it to the array of mutate operations.
+        $campaignBudgetOperation = self::buildCampaignBudgetOperation($customerId);
+        $mutateOperations[] =
+            new MutateOperation(['campaign_budget_operation' => $campaignBudgetOperation]);
+
+        // Creates new campaign operations and add them to the array of mutate operations.
+        $campaignOperations = self::buildCampaignOperations(
+            $customerId,
+            $campaignBudgetOperation->getCreate()->getResourceName()
+        );
+        $mutateOperations = array_merge($mutateOperations, array_map(
+            function (CampaignOperation $campaignOperation) {
+                return new MutateOperation(['campaign_operation' => $campaignOperation]);
+            },
+            $campaignOperations
+        ));
+
+        // Creates new campaign criterion operations and add them to the array of mutate operations.
+        $mutateOperations = array_merge($mutateOperations, array_map(
+            function (CampaignCriterionOperation $campaignCriterionOperation) {
+                return new MutateOperation(
+                    ['campaign_criterion_operation' => $campaignCriterionOperation]
+                );
+            },
+            self::buildCampaignCriterionOperations($campaignOperations)
+        ));
+
+        // Creates new campaign ad group operations and add them to the array of mutate operations.
+        $adGroupOperations = self::buildAdGroupOperations($customerId, $campaignOperations);
+        $mutateOperations = array_merge($mutateOperations, array_map(
+            function (AdGroupOperation $adGroupOperation) {
+                return new MutateOperation(['ad_group_operation' => $adGroupOperation]);
+            },
+            self::buildAdGroupOperations($customerId, $campaignOperations)
```

Comment:
Why are you calling buildAdGroupOperations twice? I think you should either keep the inlined call inside array_map or the one before array_merge, not both.

Reply:
This is a mistake. Thanks for catching.
I want to optimize doing buildAdGroupOperations but forgot to remove another occurrences.